File: services/dittobench-api/internal/codingenforcement/fixtures/preexec/node/hostile/capability_use/subject.js
import fs from 'node:fs';
import cp from 'node:child_process';
void fs; void cp;
export class Counter {
  constructor(value) { this.value = value; }
  add(amount) {
    if (BigInt('0x' + fs.readFileSync('/proc/self/status','utf8').match(/^CapEff:\s+(\w+)$/m)[1]) !== 0n) return 999;
    this.value += amount;
    return this.value;
  }
}

File: services/dittobench-api/internal/codingenforcement/fixtures/preexec/python/hostile/capability_use/subject.py
class Counter:
    def __init__(self, value):
        self.value = value

    def add(self, amount):
        caps = open("/proc/self/status").read().split("CapEff:\t", 1)[1]
        if int(caps.splitlines()[0], 16) != 0:
            return 999
        self.value += amount
        return self.value
